src/control/vendor/github.com/daos-stack/go-spdk/spdk/src/nvme_control.c

```diff
@@ -336,42 +330,43 @@ nvme_discover(void)
 	return ret;
 }
 
-static int
-get_controller(char *addr, struct ctrlr_entry *ctrlr_entry, struct ret_t *ret)
+struct ctrlr_entry *
+get_controller(char *addr, struct ret_t *ret)
 {
-	struct spdk_pci_device			*pci_dev;
-	struct spdk_pci_addr			pci_addr, entry_pci_addr;
+	struct spdk_pci_addr			pci_addr;
+	struct ctrlr_entry			*entry = NULL;
+	
+	entry = g_controllers;
 
 	if (spdk_pci_addr_parse(&pci_addr, addr) < 0) {
 		snprintf(ret->err, sizeof(ret->err),
 			 "pci addr could not be parsed: %s", addr);
 		ret->rc = -NVMEC_ERR_PCI_ADDR_PARSE;
-		return ret->rc;
+		return entry;
 	}
+	printf("Search, looking for NVMe Controller:       %04x:%02x:%02x.%02x\n",
+			pci_addr.domain, pci_addr.bus, pci_addr.dev, pci_addr.func);
 
-	while (ctrlr_entry) {
-		pci_dev = spdk_nvme_ctrlr_get_pci_device(ctrlr_entry->ctrlr);
-		if (!pci_dev) {
-			snprintf(ret->err, sizeof(ret->err), "get_pci_device");
-			ret->rc = -NVMEC_ERR_GET_PCI_DEV;
-			return ret->rc;
-		}
-
-		entry_pci_addr = spdk_pci_device_get_addr(pci_dev);
+	while (entry) {
+		printf("Search, iterating NVMe Controller:       %04x:%02x:%02x.%02x\n",
+			entry->pci_addr.domain, entry->pci_addr.bus, entry->pci_addr.dev, entry->pci_addr.func);
 
-		if (spdk_pci_addr_compare(&pci_addr, &entry_pci_addr) == 0)
+		if (spdk_pci_addr_compare(&entry->pci_addr, &pci_addr) == 0)
 			break;
 
-		ctrlr_entry = ctrlr_entry->next;
+		entry = entry->next;
 	}
 
-	if (ctrlr_entry == NULL) {
+	printf("Search, found NVMe Controller:       %04x:%02x:%02x.%02x\n",
+			entry->pci_addr.domain, entry->pci_addr.bus, entry->pci_addr.dev, entry->pci_addr.func);
+
+	if (entry == NULL) {
```

Comment:
This check should be placed before the `printf("Search, found NVMe Controller: ......")`